src/common/ceph_clock.h

```diff
@@ -0,0 +1,187 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * Author: Shinobu Kinjo <shinobu@redhat.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+#ifndef CEPH_CLOCK_H
+#define CEPH_CLOCK_H
+
+#include "common/ceph_time.h"
+#include "include/utime.h"
+#include <type_traits>
+#include <time.h>
+
+using namespace std::chrono;
+
+struct ceph_clock {
+  ceph_clock() :
+    mc_start_(ceph::mono_clock::now()),
+    cc_start_(ceph::coarse_mono_clock::now()),
+    ut_start_(ceph_clock_now_()) {} 
```

Comment:
i really don't think we want to read from all 3 clocks on construction

Reply:
the purpose of distinguishing starting time from construction time is that users of this wrapper should have responsibility of starting point of calculating latency.

we should be accurate as much as possible when it comes to latency.